usaspending_api/references/models.py

```diff
@@ -401,10 +470,303 @@ class LegalEntity(DataSourceTrackedModel):
     individual = models.TextField(blank=True, null=True)
 
     def save(self, *args, **kwargs):
+        LegalEntity.update_business_type_categories(self)
         super(LegalEntity, self).save(*args, **kwargs)
 
         LegalEntityOfficers.objects.get_or_create(legal_entity=self)
 
+    @staticmethod
+    def update_business_type_categories(le):
+        # Business Category
+        le.category_small_business = (
+            le.small_business == "1" or
+
+            le.business_types == "R"  # For-Profit Organization (Other than Small Business)
+        )
+        le.category_other_than_small_business = (
+            le.business_types == "Q"  # For-Profit Organization (Other than Small Business)
+        )
+        le.category_business = (
+            le.for_profit_organization == "1" or
+
+            le.category_small_business or
+            le.category_other_than_small_business
+        )
+        # End Business Category
+
+        # Minority Owned Business Category
+        le.category_alaskan_native_owned_business = (
+            le.alaskan_native_owned_corporation_or_firm == "1"
```

Comment:
These are all (so far) coming from the Broker, so they are all going to be the same (1 or 0 or null (since the source field in the broker is boolean)) and the business flags are a similar case. When we load historical items we'll need to ensure they're properly formatted, true.